app/uk/gov/hmrc/gform/testonly/SnapshotADTs.scala

```diff
@@ -16,78 +16,166 @@
 
 case class Snapshot(
-  templateId: String,
-  snapshotId: String,
+  templateId: FormTemplateId,
+  snapshotId: SnapshotId,
   savedAt: Instant,
-  description: String
+  description: Description,
+  gformVersion: GformVersion,
+  gformFrontendVersion: GformFrontendVersion
 )
 
 object Snapshot {
 
-  def apply(cacheItem: CacheItem): Snapshot = {
-    val templateId = cacheItem.data.validate((__ \ "form" \ "formTemplateId").json.pick[JsString]).get.value
-    val snapshotId = cacheItem.id
-    val savedAt = cacheItem.createdAt
-    val description = cacheItem.data.validate((__ \ "description").json.pick[JsString]).get.value
+  def apply(snapshotItem: SnapshotItem): Snapshot =
     Snapshot(
-      templateId,
-      snapshotId,
-      savedAt,
-      description
+      snapshotItem.snapshotTemplateId,
+      snapshotItem.snapshotId,
+      snapshotItem.createdAt,
+      snapshotItem.description,
+      snapshotItem.gformVersion,
+      snapshotItem.gformFrontendVersion
     )
-  }
 
   implicit val writes: OWrites[Snapshot] = derived.owrites()
-
 }
 
 case class SnapshotWithData(
   snapshot: Snapshot,
-  formData: JsObject
+  formData: FormData
 )
 object SnapshotWithData {
+  def apply(snapshotItem: SnapshotItem): SnapshotWithData =
+    SnapshotWithData(Snapshot(snapshotItem), snapshotItem.originalForm.formData)
   implicit val writes: OWrites[SnapshotWithData] = derived.owrites()
 }
 
 case class SaveRequest(
-  formId: String,
-  description: String
+  formId: FormId,
+  description: Description,
+  gformFrontendVersion: GformFrontendVersion
 )
 
 object SaveRequest {
   implicit val format: OFormat[SaveRequest] = derived.oformat()
 }
 
 case class SaveReply(
-  snapshotId: String
+  formId: FormId
 )
 
 object SaveReply {
   implicit val format: OFormat[SaveReply] = derived.oformat()
 }
 
 case class UpdateSnapshotRequest(
-  snapshotId: String,
+  snapshotId: SnapshotId,
   formData: JsObject,
-  description: String
+  description: Description
 )
 
 object UpdateSnapshotRequest {
   implicit val format: OFormat[UpdateSnapshotRequest] = derived.oformat()
 }
 
 case class UpdateFormDataRequest(
-  formId: String,
+  formId: FormId,
   formData: JsObject
 )
 
 object UpdateFormDataRequest {
   implicit val format: OFormat[UpdateFormDataRequest] = derived.oformat()
 }
+case class SnapshotId(value: String) extends AnyVal
+
+object SnapshotId {
+  implicit val format: Format[SnapshotId] = Json.valueFormat
+}
+
+case class Description(value: String) extends AnyVal
+object Description {
+  implicit val format: Format[Description] = Json.valueFormat
+}
+
+case class GformVersion(value: String) extends AnyVal
+
+object GformVersion {
+  implicit val format: Format[GformVersion] = Json.valueFormat
+}
+
+case class GformFrontendVersion(value: String) extends AnyVal
+
+object GformFrontendVersion {
+  implicit val format: Format[GformFrontendVersion] = Json.valueFormat
+}
+
+case class SnapshotItem(
```

Comment:
It would be nice to call this simply `Snapshot`

Reply:
but how would I call current `Snapshot` that is used for List[Snapshot]

Reply:
Shall I rename `Snapshot` to `SnapshotMeta` and `SnapshotItem` to `Snapshot`?

Reply:
We can call `SnapshotOverview` the class to be used in a List.